nmagent/equality.go

```diff
@@ -0,0 +1,71 @@
+package nmagent
+
+// equalPtr compares two Interfaces objects for equality.
+func (i *Interfaces) equalPtr(other *Interfaces) bool {
+	if len(i.Entries) != len(other.Entries) {
+		return false
+	}
+	for idx, entry := range i.Entries {
+		if !entry.equalPtr(&other.Entries[idx]) {
+			return false
+		}
+	}
+	return true
+}
+
+// Equal compares two Interfaces objects for equality.
+func (i Interfaces) Equal(other Interfaces) bool {
+	return i.equalPtr(&other)
+}
```

Comment:
I don't think this double implementation should be necessary due to automatic referencing / dereferencing. @rbtr ?

Reply:
I believe the automatic dereferencing works for i (in fact, you can see I didn't do `&i.equalPtr`, but he `other` parameter needs to be explicitly referenced I think.

Reply:
@santhoshmprabhu did you try with implementing only the pointer receiver on these and see if it breaks?

Reply:
@timraymond Yup. The pointer version needs to be explicitly called with the `(&other)` param. We could of course have the client return pointers rather than the struct itself and then work entirely with pointers. But then that somewhat breaks from the existing pattern of structs and umarshaling in the client. Do you want me to do that?

Reply:
This passes your tests: https://github.com/Azure/azure-container-networking/commit/1fbfb6e7c97d8a0630b8f5ffe58b0165bab363bf

Reply:
It does, but it copies objects back and forth several times. Each call to `Equal` creates copies of the two things you're trying to compare. You can see I had that first, explicitly switched to pointers in this commit:
https://github.com/Azure/azure-container-networking/pull/3023/commits/2331766ea80a04732880a9ebc9bd20f245e33dbe

Reply:
Fair, but unless we have a measurable performance impact, code readability is the priority over optimizations. If `Equal` shows up as a big blob on a pprof, then we know what optimization to take. Until it does though, we don't know if we're compromising readability for little gain.

Reply:
Switched to the non-pointer versions only.